lib/ansible/modules/network/iosxr/iosxr_config.py

```diff
@@ -260,7 +261,8 @@ def run(module, result):
 
             result['updates'] = commands
 
-        load_config(module, commands, result)
+        load_config(module, commands, not module.check_mode)
+        result['changed'] = not module.check_mode
```

Comment:
Also, if result should return the diff key with the diff in it to support calling `ansible-playbook` with the `--diff` option